### ISSUE TITLE
Use external tool for primer trimming

### ISSUE THREAD

**Issue by peterjc** [peterjc/thapbi-pict#82]
We know the expected left and right ends of the ITS1 sequences we are expecting from the primers used, and trim them off - see #45.

Currently we call trimmomatic with the Illumina ``TruSeq3-PE.fa`` adapters. Can we tell it the primer sequences instead? This might cope nicely with possible read-though where the amplified region happens to be shorter than the MiSeq reads.

**Comment by peterjc** [comment]
Even the ``DNAMIX_S95_L001`` test dataset on the repository has some test cases, examples where the right primer is there, but not right at the end of the merged read.

**Comment by peterjc** [comment]
Getting the primer FASTA file correct seems tricky, 16 pairs didn't work as expected:

```python
>>> from thapbi_pict.utils import expand_IUPAC_ambiguity_codes
>>> f = "GAAGGTGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTA"
>>> with open("primers.fa", "w") as handle:
...     for i, r in enumerate(expand_IUPAC_ambiguity_codes("GCARRGACTTTCGTCCCYRC")):
...         handle.write(">primer%i/1\n%s\n>primer%i/2\n%s\n" % (i+1, f, i+1, r))
...
```

1 forward and 16 back:

```python
>>> from thapbi_pict.utils import expand_IUPAC_ambiguity_codes
>>> f = "GAAGGTGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTA"
>>> with open("primers.fa", "w") as handle:
.. .    handle.write(">left/1\n%s\n" % f)
...     for i, r in enumerate(expand_IUPAC_ambiguity_codes("GCARRGACTTTCGTCCCYRC")):
...         handle.write(">right%i/2\n%s\n" % (i+1, r))
...
```

In both cases trimmomatic crashes.

**Comment by peterjc** [comment]
Can do trimmomatic -> cutadapt -> pear, using paired end mode:

```
$ cutadapt -g GAAGGTGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTA -G GCARRGACTTTCGTCCCYRC -o trimmed_R1.fastq -p trimmed_R1.fastq input_R1.fastq input_R2.fastq
```

Seems doing trimmomatic -> pear -> cutadapt works better, here using non-anchored linked primers:

```
$ cutadapt -g GAAGGTGAAGTCGTAACAAGGTTTCCGTAGGTGAACCTGCGGAAGGATCATTA...GYRGGGACGAAAGTCYYTGC -o ./trimmomatic.pear.cutadapt.fasta ./trimmomatic.pear.assembled.fastq
```


In terms of the number of HMM trimming warnings (added for #43), using the ``DNAMIX_S95_L001`` test sample:

trimmomatic -> cutadapt -> pear, 19 warnings - 1 left, 16 right, 2 both
trimmomatic -> pear -> cutadapt, 11 warnings - 1 left, 10 right
trimmomatic -> pear -> fixed-trim, 3 warnings - 3 right
trimmomatic -> pear -> fuzzy-trim, same 3 warnings https://github.com/peterjc/thapbi-pict/pull/83

The left warning when using cutadapt looks like an example with 2 substitutions in the left primer, which with the defaults is therefore not a match, and not removed. Meanwhile our Python script will trim it anyway.

Likewise the right HMM warnings will likely include poor primer matches, which the fixed trimming removes anyway.

**Comment by peterjc** [comment]
There is a case for discarding reads where the primers do not match, e.g. via ``cutadapt --discard-untrimmed ...``

Or, try changing the default error rate with ``cutadapt -e 0.1 ...`` option (default ``0.1`` meaning 10%)

**Comment by peterjc** [comment]
Plan agreed with @widdowquinn, discard reads where primers not found, but warn if they had high abundance. Will probably need to use this option with cutadapt:

```
    --untrimmed-output=FILE
                        Write reads that do not contain any adapter to FILE.
                        Default: output to same file as trimmed reads
```

**Comment by peterjc** [comment]
Fixed via 20ed57d128d69291118f308838b7d11c40b7afe6 in v0.0.10

Detailed comparison to v0.0.9 where I did primer detection in Python allowing 1bp differences, cutadapt allows 2bp and thus finds more primers and accepts more sequences. This results in slightly higher abundance counts, but we only see a non-trivial change in the sequences prepared on plates with a relatively low minimum abundance threshold. In particular, there is no change on our control plate.

Comparison to v0.0.8a where I did fixed left/right trimming without primer detection is more complicated. However, in terms of the control plate, at sample level the ``swarm`` and ``swarmid`` classifiers now get ``Phytophthora fallax`` correct, but otherwise unchanged. Comparison at sequence level is problematic as the number of accepted sequences is different - the old code accepts far more sequences (those with primer problems).
